js/ui/date_box/ui.date_box.mask.js

```diff
@@ -96,7 +96,9 @@ const DateBoxMask = DateBoxBase.inherit({
     },
 
     _shouldUseOriginalHandler(e) {
-        return !this._useMaskBehavior() || this.option("opened") || (e && e.altKey);
+        const keysToHandleByMask = ["backspace", "del"];
+
+        return !this._useMaskBehavior() || (this.option("opened") && e && !keysToHandleByMask.includes(normalizeKeyName(e))) || (e && e.altKey);
```

Comment:
```suggestion
        return !this._useMaskBehavior() || (this.option("opened") && e && keysToHandleByMask.indexOf(normalizeKeyName(e))) === -1 || (e && e.altKey);
```

IE11 doesn't support [Array.prototype.includes](https://caniuse.com/#feat=array-includes)